tests/python/pants_test/util/test_meta.py

```diff
@@ -246,6 +252,21 @@ def f(cls):
     self.assertEqual(Concrete2.f, 'hello')
 
 
+class SentinelAttributeTest(unittest.TestCase):
+
+  def test_decorated_type_checkable(self):
+    @decorated_type_checkable
+    def f(cls):
+      return f.define_instance_of(cls)
+
+    @f
+    class C:
+      pass
+
+    self.assertEqual(C._decorated_type_checkable_type, type(f))
+    self.assertTrue(f.is_instance(C))
```

Comment:
One tiny last thing: Maybe a check that a different class which is decorated with a _different_ decorator (`@g`) is an instance of `g` but not `f`? Then I'm done, let's merge!